components/formats-gpl/src/loci/formats/in/CV7000Reader.java

```diff
@@ -0,0 +1,900 @@
+          }
+          LOGGER.error("Error parsing attributes: {}", attributeMap, e);
+        }
+        throw e;
+      }
+    }
+
+    @Override
+    public void endElement(String uri, String localName, String qName) {
+      String value = currentValue.toString();
+      if (qName.equals("bts:MeasurementRecord") && btsType.equals("IMG") &&
+        value.trim().length() > 0) {
+        Location imgFile = new Location(parentDir, value);
+        if (imgFile.exists()) {
+          planes.get(planes.size() - 1).file = imgFile.getAbsolutePath();
+        }
+      }
+    }
+
+  }
+
+  class MeasurementDetailHandler extends BaseHandler {
+
+    // -- DefaultHandler API methods --
+
+    @Override
+    public void startElement(String uri, String localName, String qName,
+      Attributes attributes)
+    {
+      if (qName.equals("bts:MeasurementSamplePlate")) {
+        wppPath = attributes.getValue("bts:WellPlateProductFileName");
+        if (wppPath != null && wppPath.trim().length() == 0) {
+          wppPath = null;
+        }
+      }
+      else if (qName.equals("bts:MeasurementChannel")) {
+        Channel c = new Channel();
+        c.index = Integer.parseInt(attributes.getValue("bts:Ch")) - 1;
+        c.xSize = Double.parseDouble(attributes.getValue("bts:HorizontalPixelDimension"));
+        c.ySize = Double.parseDouble(attributes.getValue("bts:VerticalPixelDimension"));
+        c.cameraNumber = Integer.parseInt(attributes.getValue("bts:CameraNumber"));
+        c.correctionFile = attributes.getValue("bts:ShadingCorrectionSource");
+        if (c.correctionFile != null && c.correctionFile.trim().length() == 0) {
+          c.correctionFile = null;
+        }
+        channels.add(c);
+      }
+      else if (qName.equals("bts:MeasurementDetail")) {
+        startTime = attributes.getValue("bts:BeginTime");
+        endTime = attributes.getValue("bts:EndTime");
+        settingsPath = attributes.getValue("bts:MeasurementSettingFileName");
+      }
+    }
+
+  }
+
+  class MeasurementSettingsHandler extends BaseHandler {
+    private Channel currentChannel = null;
+    private StringBuffer currentValue = new StringBuffer();
+
+    // -- DefaultHandler API methods --
+
+    @Override
+    public void characters(char[] ch, int start, int length) {
+      String value = new String(ch, start, length);
+      currentValue.append(value);
+    }
+
+    @Override
+    public void startElement(String uri, String localName, String qName,
+      Attributes attributes)
+    {
+      currentValue.setLength(0);
+      if (qName.equals("bts:LightSource")) {
+        LightSource l = new LightSource();
+        l.name = attributes.getValue("bts:Name");
+        l.type = attributes.getValue("bts:Type");
+
+        String wavelength = attributes.getValue("bts:WaveLength");
+        String power = attributes.getValue("bts:Power");
+
+        if (wavelength != null) {
+          try {
+            l.wavelength = new Double(wavelength);
```

Comment:
Could we use `Double.parseDouble` for these?

Reply:
a49b94b switches all usage of ```new Double``` and ```Double.parseDouble``` to ```DataTools.parseDouble```.